Static library build on windows seems to have issues on Andre's machine

I can build and install the static libOpenCOR with MS Visual Studio 22. With the example `ccptest` that the build instructions recommend to use to test the installation, I can manually give the path to the static library and header files and the app builds fine and runs to write out the libOpenCOR version. I can also do this in my own application. But when I try to use other functionality (i.e., load a file), I get loads of linking errors that seem to be mostly to do with things that look like SSL.

I also grabbed pre-built libraries from @agarny (which work fine for him), and get the same result.

Just to confirm that, following the instructions at https://opencor.ws/libopencor/user/installation/fromSource.html, I cannot build `cpptest` using the static version of libOpenCOR:
```
C:\Users\Alan\Desktop\cpptest> cmake -S . -B build -A x64 -DCMAKE_PREFIX_PATH="C:\Program Files\libOpenCOR\cmake"
-- Building for: Visual Studio 17 2022
-- Selecting Windows SDK version 10.0.22621.0 to target Windows 10.0.26100.
-- The C compiler identification is MSVC 19.42.34436.0
-- The CXX compiler identification is MSVC 19.42.34436.0
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: C:/Program Files/Microsoft Visual Studio/2022/Community/VC/Tools/MSVC/14.42.34433/bin/Hostarm64/x64/cl.exe - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: C:/Program Files/Microsoft Visual Studio/2022/Community/VC/Tools/MSVC/14.42.34433/bin/Hostarm64/x64/cl.exe - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Configuring done (2.7s)
CMake Error at C:/Program Files/libOpenCOR/cmake/libOpenCOR/libOpenCORTargets.cmake:60 (set_target_properties):
  The link interface of target "libOpenCOR" contains:

    OpenSSL::libssl

  but the target was not found.  Possible reasons include:

    * There is a typo in the target name.
    * A find_package call is missing for an IMPORTED target.
    * An ALIAS target is missing.

Call Stack (most recent call first):
  C:/Program Files/libOpenCOR/cmake/libOpenCOR/libOpenCORConfig.cmake:41 (include)
  CMakeLists.txt:5 (find_package)


-- Generating done (0.0s)
CMake Generate step failed.  Build files cannot be regenerated correctly.
```
As expected, it's related to the way libOpenCOR's static version is now built (see issue #385). As a result, the call to `set_target_properties()` in `<InstallDir>\libOpenCOR\cmake\libOpenCOR\libOpenCORTargets.cmake` shouldn't have the `INTERFACE_LINK_LIBRARIES` line anymore.

Thanks - that lets the build work and I can use `FIND_PACKAGE` in my project to find and build against the installed libOpenCOR (under windows). But now I'm getting lots of link unresolved symbols errors as shown below. This is from Visual Studio Code doing some magic to somehow build things for me. I'll see if I can do more manual testing to see if that helps. All the test `.exe` files built run fine and succeed.

```
[build] libOpenCOR.lib(libcrypto-lib-bio_sock.obj) : error LNK2001: unresolved external symbol __imp_getsockopt [C:\Users\dnic019\development\libopencor\csim2\build\csim2.vcxproj]
[build] libOpenCOR.lib(libcrypto-lib-bio_sock2.obj) : error LNK2001: unresolved external symbol __imp_getsockopt [C:\Users\dnic019\development\libopencor\csim2\build\csim2.vcxproj]
[build] libOpenCOR.lib(multi.c.obj) : error LNK2001: unresolved external symbol __imp_getsockopt [C:\Users\dnic019\development\libopencor\csim2\build\csim2.vcxproj]
[build] libOpenCOR.lib(nanohttp.c.obj) : error LNK2001: unresolved external symbol __imp_getsockopt [C:\Users\dnic019\development\libopencor\csim2\build\csim2.vcxproj]
[build] libOpenCOR.lib(session.c.obj) : error LNK2001: unresolved external symbol __imp_getsockopt [C:\Users\dnic019\development\libopencor\csim2\build\csim2.vcxproj]
[build] libOpenCOR.lib(cf-socket.c.obj) : error LNK2001: unresolved external symbol __imp_getsockopt [C:\Users\dnic019\development\libopencor\csim2\build\csim2.vcxproj]
[build] libOpenCOR.lib(libcrypto-lib-bss_sock.obj) : error LNK2001: unresolved external symbol __imp_send [C:\Users\dnic019\development\libopencor\csim2\build\csim2.vcxproj]
[build] libOpenCOR.lib(libcrypto-lib-bss_conn.obj) : error LNK2001: unresolved external symbol __imp_send [C:\Users\dnic019\development\libopencor\csim2\build\csim2.vcxproj]
[build] libOpenCOR.lib(socketpair.c.obj) : error LNK2001: unresolved external symbol __imp_send [C:\Users\dnic019\development\libopencor\csim2\build\csim2.vcxproj]
[build] libOpenCOR.lib(telnet.c.obj) : error LNK2001: unresolved external symbol __imp_send [C:\Users\dnic019\development\libopencor\csim2\build\csim2.vcxproj]
[build] libOpenCOR.lib(cf-socket.c.obj) : error LNK2001: unresolved external symbol __imp_send [C:\Users\dnic019\development\libopencor\csim2\build\csim2.vcxproj]
[build] libOpenCOR.lib(misc.c.obj) : error LNK2001: unresolved external symbol __imp_send [C:\Users\dnic019\development\libopencor\csim2\build\csim2.vcxproj]
[build] libOpenCOR.lib(multi.c.obj) : error LNK2001: unresolved external symbol __imp_send [C:\Users\dnic019\development\libopencor\csim2\build\csim2.vcxproj]
....
....
....
[build] libOpenCOR.lib(libcrypto-lib-e_capi.obj) : error LNK2019: unresolved external symbol __imp_CertDuplicateCertificateContext referenced in function capi_load_ssl_client_cert [C:\Users\dnic019\development\libopencor\csim2\build\csim2.vcxproj]
[build] libOpenCOR.lib(libcrypto-lib-e_capi.obj) : error LNK2019: unresolved external symbol __imp_CertGetCertificateContextProperty referenced in function capi_cert_get_fname [C:\Users\dnic019\development\libopencor\csim2\build\csim2.vcxproj]
[build] libOpenCOR.lib(libcrypto-lib-bss_conn.obj) : error LNK2019: unresolved external symbol __imp_shutdown referenced in function conn_free [C:\Users\dnic019\development\libopencor\csim2\build\csim2.vcxproj]
[build] C:\Users\dnic019\development\libopencor\csim2\build\Release\csim2.exe : fatal error LNK1120: 74 unresolved externals [C:\Users\dnic019\development\libopencor\csim2\build\csim2.vcxproj]
[proc] The command: "C:\Program Files\CMake\bin\cmake.exe" --build c:/Users/dnic019/development/libopencor/csim2/build --config Release --target ALL_BUILD -j 10 -- exited with code: 1
[driver] Build completed: 00:00:14.132
[build] Build finished with exit code 1
```

--- a/.github/workflows/buildThirdPartyLibrary.yml
+++ b/.github/workflows/buildThirdPartyLibrary.yml
@@ -42,9 +42,8 @@ jobs:
             os: ubuntu-22.04-arm
             build_type: Release
           - name: 'macOS (Intel)'
-            os: macos-14
+            os: macos-13
             build_type: Release
-            target_arch: " -DTARGET_ARCHITECTURE=Intel"
           - name: 'macOS (ARM)'
             os: macos-14
             build_type: Release
@@ -76,10 +75,10 @@ jobs:
       - name: Configure libOpenCOR (only when building LLVM+Clang)
         if: ${{ inputs.third_party_library_name == 'LLVMClang' }}
         shell: bash
-        run: cmake -G Ninja -S . -B build ${{ matrix.target_arch }} -DBUILD_TYPE=${{ matrix.build_type }} -DONLY_BUILD_THIRD_PARTY_LIBRARIES=ON -DPREBUILT_LLVMCLANG=OFF
+        run: cmake -G Ninja -S . -B build -DBUILD_TYPE=${{ matrix.build_type }} -DONLY_BUILD_THIRD_PARTY_LIBRARIES=ON -DPREBUILT_LLVMCLANG=OFF
       - name: Configure libOpenCOR (only when NOT building LLVM+Clang)
         if: ${{ inputs.third_party_library_name != 'LLVMClang' }}
-        run: cmake -G Ninja -S . -B build ${{ matrix.target_arch }} -DBUILD_TYPE=${{ matrix.build_type }} -DONLY_BUILD_THIRD_PARTY_LIBRARIES=ON -DPREBUILT_${{ inputs.third_party_library_name }}=OFF
+        run: cmake -G Ninja -S . -B build -DBUILD_TYPE=${{ matrix.build_type }} -DONLY_BUILD_THIRD_PARTY_LIBRARIES=ON -DPREBUILT_${{ inputs.third_party_library_name }}=OFF
       - name: Upload library artifact
         if: ${{ !startsWith(github.ref, 'refs/tags/v') }}
         uses: actions/upload-artifact@v4

--- a/.github/workflows/buildThirdPartyLibrary.yml
+++ b/.github/workflows/buildThirdPartyLibrary.yml
@@ -76,16 +76,10 @@ jobs:
       - name: Configure libOpenCOR (only when building LLVM+Clang)
         if: ${{ inputs.third_party_library_name == 'LLVMClang' }}
         shell: bash
-        run: |
-          mkdir build
-          cd build
-          cmake -G Ninja ${{ matrix.target_arch }} -DBUILD_TYPE=${{ matrix.build_type }} -DONLY_BUILD_THIRD_PARTY_LIBRARIES=ON -DPREBUILT_LLVMCLANG=OFF ..
+        run: cmake -G Ninja -S . -B build ${{ matrix.target_arch }} -DBUILD_TYPE=${{ matrix.build_type }} -DONLY_BUILD_THIRD_PARTY_LIBRARIES=ON -DPREBUILT_LLVMCLANG=OFF
       - name: Configure libOpenCOR (only when NOT building LLVM+Clang)
         if: ${{ inputs.third_party_library_name != 'LLVMClang' }}
-        run: |
-          mkdir build
-          cd build
-          cmake -G Ninja ${{ matrix.target_arch }} -DBUILD_TYPE=${{ matrix.build_type }} -DONLY_BUILD_THIRD_PARTY_LIBRARIES=ON -DPREBUILT_${{ inputs.third_party_library_name }}=OFF ..
+        run: cmake -G Ninja -S . -B build ${{ matrix.target_arch }} -DBUILD_TYPE=${{ matrix.build_type }} -DONLY_BUILD_THIRD_PARTY_LIBRARIES=ON -DPREBUILT_${{ inputs.third_party_library_name }}=OFF
       - name: Upload library artifact
         if: ${{ !startsWith(github.ref, 'refs/tags/v') }}
         uses: actions/upload-artifact@v4
@@ -117,14 +111,9 @@ jobs:
       - name: Install Emscripten
         run: brew install emscripten
       - name: Configure libOpenCOR
-        run: |
-          mkdir build
-          cd build
-          cmake -G Ninja -DBUILD_TYPE=Release -DJAVASCRIPT_BINDINGS=ON -DONLY_BUILD_THIRD_PARTY_LIBRARIES=ON -DPREBUILT_${{ inputs.third_party_library_name }}=OFF ..
+        run: cmake -G Ninja -S . -B build -DBUILD_TYPE=Release -DJAVASCRIPT_BINDINGS=ON -DONLY_BUILD_THIRD_PARTY_LIBRARIES=ON -DPREBUILT_${{ inputs.third_party_library_name }}=OFF
       - name: Build WebAssembly
-        run: |
-          cd build
-          ninja
+        run: cmake --build build
       - name: Upload WebAssembly artifact
         if: ${{ !startsWith(github.ref, 'refs/tags/v') }}
         uses: actions/upload-artifact@v4

--- a/.github/workflows/cd.yml
+++ b/.github/workflows/cd.yml
@@ -171,13 +171,18 @@ jobs:
         uses: pypa/cibuildwheel@v2.23.3
         env:
           CIBW_ARCHS: auto64
+          CIBW_BEFORE_ALL_LINUX: >
+            export USR_LIB_PATH=/usr/lib/`uname -m`-linux-gnu &&
+            mkdir -p $USR_LIB_PATH &&
+            cd $USR_LIB_PATH &&
+            ln -s /usr/lib64/libz.so
           CIBW_BUILD: 'cp*'
           CIBW_ENVIRONMENT_MACOS: MACOSX_DEPLOYMENT_TARGET=13.0
           # Note #1: the value of CIBW_ENVIRONMENT_MACOS must be in sync with that of MACOS_DEPLOYMENT_TARGET in
           #          CMakelists.txt.
           # Note #2: for some reasons, we need to specify the minor version, hence using 13.0 instead of 13.
-          CIBW_MANYLINUX_X86_64_IMAGE: manylinux_2_34
-          CIBW_MANYLINUX_AARCH64_IMAGE: manylinux_2_34
+          CIBW_MANYLINUX_X86_64_IMAGE: quay.io/pypa/manylinux_2_34_x86_64
+          CIBW_MANYLINUX_AARCH64_IMAGE: quay.io/pypa/manylinux_2_34_aarch64
           CIBW_SKIP: 'cp36* cp37* cp38* *musllinux*'
       - name: Upload Python wheel artifacts
         if: ${{ !startsWith(github.ref, 'refs/tags/v') }}

--- a/.github/workflows/cd.yml
+++ b/.github/workflows/cd.yml
@@ -168,7 +168,7 @@ jobs:
         with:
           arch: ${{ matrix.arch }}
       - name: Build Python wheels
-        uses: pypa/cibuildwheel@9c75ea15c2f31a77e6043b80b1b7081372319d85
+        uses: pypa/cibuildwheel@v2.23.3
         env:
           CIBW_ARCHS: auto64
           CIBW_BUILD: 'cp*'

--- a/.github/workflows/cd.yml
+++ b/.github/workflows/cd.yml
@@ -109,14 +109,9 @@ jobs:
         with:
           arch: ${{ matrix.arch }}
       - name: Configure libOpenCOR
-        run: |
-          mkdir build
-          cd build
-          cmake -G Ninja ${{ matrix.target_arch }} -DBUILD_TYPE=${{ matrix.build_type }} -DCODE_ANALYSIS=OFF -DCODE_COVERAGE=OFF -DDOCUMENTATION=OFF -DJAVASCRIPT_BINDINGS=OFF -DMEMORY_CHECKS=OFF -DPYTHON_BINDINGS=OFF -DPYTHON_UNIT_TESTING=OFF -DSHARED_LIBS=${{ matrix.shared_libs }} -DUNIT_TESTING=OFF ..
+        run: cmake -G Ninja -S . -B build ${{ matrix.target_arch }} -DBUILD_TYPE=${{ matrix.build_type }} -DCODE_ANALYSIS=OFF -DCODE_COVERAGE=OFF -DDOCUMENTATION=OFF -DJAVASCRIPT_BINDINGS=OFF -DMEMORY_CHECKS=OFF -DPYTHON_BINDINGS=OFF -DPYTHON_UNIT_TESTING=OFF -DSHARED_LIBS=${{ matrix.shared_libs }} -DUNIT_TESTING=OFF
       - name: Build libOpenCOR
-        run: |
-          cd build
-          ninja
+        run: cmake --build build
       - name: Package libOpenCOR
         run: |
           cd build
@@ -219,14 +214,9 @@ jobs:
       - name: Install Emscripten
         run: brew install emscripten
       - name: Configure libOpenCOR
-        run: |
-          mkdir build
-          cd build
-          cmake -G Ninja -DBUILD_TYPE=Release -DCODE_ANALYSIS=OFF -DCODE_COVERAGE=OFF -DDOCUMENTATION=OFF -DJAVASCRIPT_BINDINGS=ON -DJAVASCRIPT_UNIT_TESTING=OFF -DMEMORY_CHECKS=OFF -DPYTHON_BINDINGS=OFF -DPYTHON_UNIT_TESTING=OFF -DSHARED_LIBS=OFF -DUNIT_TESTING=OFF ..
+        run: cmake -G Ninja -S . -B build -DBUILD_TYPE=Release -DCODE_ANALYSIS=OFF -DCODE_COVERAGE=OFF -DDOCUMENTATION=OFF -DJAVASCRIPT_BINDINGS=ON -DJAVASCRIPT_UNIT_TESTING=OFF -DMEMORY_CHECKS=OFF -DPYTHON_BINDINGS=OFF -DPYTHON_UNIT_TESTING=OFF -DSHARED_LIBS=OFF -DUNIT_TESTING=OFF
       - name: Build WebAssembly
-        run: |
-          cd build
-          ninja
+        run: cmake --build build
       - name: Upload WebAssembly artifact
         if: ${{ !startsWith(github.ref, 'refs/tags/v') }}
         uses: actions/upload-artifact@v4

--- a/.github/workflows/cd.yml
+++ b/.github/workflows/cd.yml
@@ -62,25 +62,19 @@ jobs:
             build_type: Release
             shared_libs: ON
           - name: 'macOS static library (Intel)'
-            os: macos-14
-            arch: Intel # Only used for buildcache's cache key.
+            os: macos-13
             build_type: Release
             shared_libs: OFF
-            target_arch: " -DTARGET_ARCHITECTURE=Intel"
           - name: 'macOS shared library (Intel)'
-            os: macos-14
-            arch: Intel # Only used for buildcache's cache key.
+            os: macos-13
             build_type: Release
             shared_libs: ON
-            target_arch: " -DTARGET_ARCHITECTURE=Intel"
           - name: 'macOS static library (ARM)'
             os: macos-14
-            arch: ARM # Only used for buildcache's cache key.
             build_type: Release
             shared_libs: OFF
           - name: 'macOS shared library (ARM)'
             os: macos-14
-            arch: ARM # Only used for buildcache's cache key.
             build_type: Release
             shared_libs: ON
     env:
@@ -102,14 +96,14 @@ jobs:
       - name: Install buildcache
         uses: opencor/buildcache-action@v1
         with:
-          cache_key: cd-${{ matrix.os }}-${{ matrix.arch }}-${{ matrix.build_type }}-${{ matrix.shared_libs }}
+          cache_key: cd-${{ matrix.os }}-${{ matrix.build_type }}-${{ matrix.shared_libs }}
       - name: Configure MSVC
         if: ${{ runner.os == 'Windows' }}
         uses: ilammy/msvc-dev-cmd@v1
         with:
           arch: ${{ matrix.arch }}
       - name: Configure libOpenCOR
-        run: cmake -G Ninja -S . -B build ${{ matrix.target_arch }} -DBUILD_TYPE=${{ matrix.build_type }} -DCODE_ANALYSIS=OFF -DCODE_COVERAGE=OFF -DDOCUMENTATION=OFF -DJAVASCRIPT_BINDINGS=OFF -DMEMORY_CHECKS=OFF -DPYTHON_BINDINGS=OFF -DPYTHON_UNIT_TESTING=OFF -DSHARED_LIBS=${{ matrix.shared_libs }} -DUNIT_TESTING=OFF
+        run: cmake -G Ninja -S . -B build -DBUILD_TYPE=${{ matrix.build_type }} -DCODE_ANALYSIS=OFF -DCODE_COVERAGE=OFF -DDOCUMENTATION=OFF -DJAVASCRIPT_BINDINGS=OFF -DMEMORY_CHECKS=OFF -DPYTHON_BINDINGS=OFF -DPYTHON_UNIT_TESTING=OFF -DSHARED_LIBS=${{ matrix.shared_libs }} -DUNIT_TESTING=OFF
       - name: Build libOpenCOR
         run: cmake --build build
       - name: Package libOpenCOR

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -394,86 +394,54 @@ jobs:
           wget https://github.com/opencor/gha/releases/download/gha/doxygen.tar.gz -O - | tar -xz
           sudo mv doxygen /usr/local/bin
       - name: Configure libOpenCOR
-        run: |
-          mkdir build
-          cd build
-          ${{ matrix.context }} cmake -G Ninja ${{ matrix.target_arch }} -DBUILD_TYPE=${{ matrix.build_type }} -DCODE_ANALYSIS=${{ matrix.code_analysis }} -DCODE_COVERAGE=${{ matrix.code_coverage }} -DDOCUMENTATION=${{ matrix.documentation }} -DINSTALL_PREFIX=${{ github.workspace }}/install -DJAVASCRIPT_BINDINGS=${{ matrix.javascript_support }} -DJAVASCRIPT_UNIT_TESTING=${{ matrix.javascript_support }} -DMEMORY_CHECKS=${{ matrix.memory_checks }} -DPYTHON_BINDINGS=${{ matrix.python_support }} -DPYTHON_UNIT_TESTING=${{ matrix.python_support }} -DSHARED_LIBS=${{ matrix.shared_libs }} -DUNIT_TESTING=${{ matrix.unit_testing }} ..
+        run: ${{ matrix.context }} cmake -G Ninja -S . -B build ${{ matrix.target_arch }} -DBUILD_TYPE=${{ matrix.build_type }} -DCODE_ANALYSIS=${{ matrix.code_analysis }} -DCODE_COVERAGE=${{ matrix.code_coverage }} -DDOCUMENTATION=${{ matrix.documentation }} -DINSTALL_PREFIX=${{ github.workspace }}/install -DJAVASCRIPT_BINDINGS=${{ matrix.javascript_support }} -DJAVASCRIPT_UNIT_TESTING=${{ matrix.javascript_support }} -DMEMORY_CHECKS=${{ matrix.memory_checks }} -DPYTHON_BINDINGS=${{ matrix.python_support }} -DPYTHON_UNIT_TESTING=${{ matrix.python_support }} -DSHARED_LIBS=${{ matrix.shared_libs }} -DUNIT_TESTING=${{ matrix.unit_testing }}
       - name: Build libOpenCOR
         if: ${{ (matrix.install_uninstall_and_package == 'ON') || (matrix.javascript_support == 'ON') || (matrix.code_analysis == 'ON') || (matrix.code_coverage == 'ON') || (matrix.memory_checks == 'ON') }}
-        run: |
-          cd build
-          ninja
+        run: cmake --build build
       - name: Unit testing
         if: ${{ ((matrix.unit_testing == 'ON') && (matrix.code_analysis == 'OFF')) || (matrix.code_coverage == 'ON') || (matrix.memory_checks == 'ON') }}
-        run: |
-          cd build
-          ninja ${{ matrix.target }}
+        run: cmake --build build --target ${{ matrix.target }}
       - name: JavaScript unit testing
         if: ${{ matrix.javascript_support == 'ON' }}
-        run: |
-          cd build
-          ninja ${{ matrix.target }}
+        run: cmake --build build --target ${{ matrix.target }}
       - name: Python unit testing
         if: ${{ matrix.python_support == 'ON' }}
-        run: |
-          cd build
-          ninja ${{ matrix.target }}
+        run: cmake --build build --target ${{ matrix.target }}
       - name: Python unit testing report
         if: ${{ matrix.python_support == 'ON' }}
-        run: |
-          cd build
-          ninja ${{ matrix.target }}_report
+        run: cmake --build build --target ${{ matrix.target }}_report
       - name: Code formatting
         if: ${{ matrix.target == 'check_code_formatting' }}
-        run: |
-          cd build
-          ninja ${{ matrix.target }}
+        run: cmake --build build --target ${{ matrix.target }}
       - name: JavaScript code formatting
         if: ${{ matrix.target == 'javascript_check_code_formatting' }}
-        run: |
-          cd build
-          ninja ${{ matrix.target }}
+        run: cmake --build build --target ${{ matrix.target }}
       - name: Python code formatting
         if: ${{ matrix.target == 'python_check_code_formatting' }}
-        run: |
-          cd build
-          ninja ${{ matrix.target }}
+        run: cmake --build build --target ${{ matrix.target }}
       - name: Code analysis
         if: ${{ matrix.name == 'Code analysis' }}
-        run: |
-          cd build
-          ninja
+        run: cmake --build build
       - name: Code coverage
         if: ${{ matrix.code_coverage == 'ON' }}
         run: |
-          cd build
-          ninja ${{ matrix.target }}
+          cmake --build build --target ${{ matrix.target }}
           if [ `ninja ${{ matrix.target }} | grep TOTAL | sed 's/ /\n/g' | grep "100.00%" | wc -l | sed 's/ //g'` -eq 4 ]; then exit 0; else exit 1; fi
       - name: Code coverage report
         if: ${{ matrix.code_coverage == 'ON' }}
-        run: |
-          cd build
-          ninja ${{ matrix.target }}_report
+        run: cmake --build build --target ${{ matrix.target }}_report
       - name: Memory checks
         if: ${{ matrix.memory_checks == 'ON' }}
-        run: |
-          cd build
-          ninja ${{ matrix.target }}
+        run: cmake --build build --target ${{ matrix.target }}
       - name: Documentation
         if: ${{ matrix.documentation == 'ON' }}
-        run: |
-          cd build
-          ninja ${{ matrix.target }}
+        run: cmake --build build --target ${{ matrix.target }}
       - name: Install libOpenCOR
         if: ${{ matrix.install_uninstall_and_package == 'ON' }}
-        run: |
-          cd build
-          ninja install
+        run: cmake --build build --target install
       - name: Uninstall libOpenCOR
         if: ${{ matrix.install_uninstall_and_package == 'ON' }}
-        run: |
-          cd build
-          ninja uninstall
+        run: cmake --build build --target uninstall
       - name: Package libOpenCOR
         if: ${{ matrix.install_uninstall_and_package == 'ON' }}
         run: |

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -439,6 +439,13 @@ jobs:
       - name: Install libOpenCOR
         if: ${{ matrix.install_uninstall_and_package == 'ON' }}
         run: cmake --build build --target install
+      - name: Test libOpenCOR
+        if: ${{ matrix.install_uninstall_and_package == 'ON' }}
+        run: |
+          cd tests/install
+          cmake -G Ninja -S . -B build
+          cmake --build build
+          build/testInstall
       - name: Uninstall libOpenCOR
         if: ${{ matrix.install_uninstall_and_package == 'ON' }}
         run: cmake --build build --target uninstall
@@ -454,7 +461,9 @@ jobs:
           pip3 install .
       - name: Test libOpenCOR Python
         if: ${{ matrix.pip_install_test_and_uninstall == 'ON' }}
-        run: python3 -c "import libopencor as oc; print(f'This is libOpenCOR version {oc.version_string()}.')"
+        run: |
+          cd tests/install/bindings/python
+          python3 test_install.py
       - name: pip uninstall libOpenCOR
         if: ${{ matrix.pip_install_test_and_uninstall == 'ON' }}
         run: pip3 uninstall -y libopencor

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -41,6 +41,7 @@ jobs:
             unit_testing: ON
             target: unit_testing
             install_uninstall_and_package: ON
+            copy_dll: ON
           - name: 'Windows static library (ARM)'
             os: windows-11-arm
             arch: amd64_arm64
@@ -69,6 +70,7 @@ jobs:
             unit_testing: ON
             target: unit_testing
             install_uninstall_and_package: ON
+            copy_dll: ON
           - name: 'Linux static library (Intel)'
             os: ubuntu-22.04
             build_type: Release
@@ -82,6 +84,7 @@ jobs:
             unit_testing: ON
             target: unit_testing
             install_uninstall_and_package: ON
+            use_install_prefix: ON
           - name: 'Linux shared library (Intel)'
             os: ubuntu-22.04
             build_type: Release
@@ -95,6 +98,7 @@ jobs:
             unit_testing: ON
             target: unit_testing
             install_uninstall_and_package: ON
+            use_install_prefix: ON
           - name: 'Linux static library (ARM)'
             os: ubuntu-22.04-arm
             build_type: Release
@@ -108,6 +112,7 @@ jobs:
             unit_testing: ON
             target: unit_testing
             install_uninstall_and_package: ON
+            use_install_prefix: ON
           - name: 'Linux shared library (ARM)'
             os: ubuntu-22.04-arm
             build_type: Release
@@ -121,6 +126,7 @@ jobs:
             unit_testing: ON
             target: unit_testing
             install_uninstall_and_package: ON
+            use_install_prefix: ON
           - name: 'macOS static library (Intel)'
             os: macos-13
             build_type: Release
@@ -134,6 +140,7 @@ jobs:
             unit_testing: ON
             target: unit_testing
             install_uninstall_and_package: ON
+            use_install_prefix: ON
           - name: 'macOS shared library (Intel)'
             os: macos-13
             build_type: Release
@@ -147,6 +154,7 @@ jobs:
             unit_testing: ON
             target: unit_testing
             install_uninstall_and_package: ON
+            use_install_prefix: ON
           - name: 'macOS static library (ARM)'
             os: macos-14
             build_type: Release
@@ -160,6 +168,7 @@ jobs:
             unit_testing: ON
             target: unit_testing
             install_uninstall_and_package: ON
+            use_install_prefix: ON
           - name: 'macOS shared library (ARM)'
             os: macos-14
             build_type: Release
@@ -173,6 +182,7 @@ jobs:
             unit_testing: ON
             target: unit_testing
             install_uninstall_and_package: ON
+            use_install_prefix: ON
           - name: 'JavaScript bindings'
             os: macos-14
             build_type: Release
@@ -392,7 +402,10 @@ jobs:
           wget https://github.com/opencor/gha/releases/download/gha/doxygen.tar.gz -O - | tar -xz
           sudo mv doxygen /usr/local/bin
       - name: Configure libOpenCOR
-        run: ${{ matrix.context }} cmake -G Ninja -S . -B build -DBUILD_TYPE=${{ matrix.build_type }} -DCODE_ANALYSIS=${{ matrix.code_analysis }} -DCODE_COVERAGE=${{ matrix.code_coverage }} -DDOCUMENTATION=${{ matrix.documentation }} -DINSTALL_PREFIX=${{ github.workspace }}/install -DJAVASCRIPT_BINDINGS=${{ matrix.javascript_support }} -DJAVASCRIPT_UNIT_TESTING=${{ matrix.javascript_support }} -DMEMORY_CHECKS=${{ matrix.memory_checks }} -DPYTHON_BINDINGS=${{ matrix.python_support }} -DPYTHON_UNIT_TESTING=${{ matrix.python_support }} -DSHARED_LIBS=${{ matrix.shared_libs }} -DUNIT_TESTING=${{ matrix.unit_testing }}
+        shell: bash
+        run: |
+          if [ '${{ matrix.use_install_prefix }}' == 'ON' ]; then export D_INSTALL_PREFIX='-DINSTALL_PREFIX=${{ github.workspace }}/install'; fi
+          ${{ matrix.context }} cmake -G Ninja -S . -B build -DBUILD_TYPE=${{ matrix.build_type }} -DCODE_ANALYSIS=${{ matrix.code_analysis }} -DCODE_COVERAGE=${{ matrix.code_coverage }} -DDOCUMENTATION=${{ matrix.documentation }} $D_INSTALL_PREFIX -DJAVASCRIPT_BINDINGS=${{ matrix.javascript_support }} -DJAVASCRIPT_UNIT_TESTING=${{ matrix.javascript_support }} -DMEMORY_CHECKS=${{ matrix.memory_checks }} -DPYTHON_BINDINGS=${{ matrix.python_support }} -DPYTHON_UNIT_TESTING=${{ matrix.python_support }} -DSHARED_LIBS=${{ matrix.shared_libs }} -DUNIT_TESTING=${{ matrix.unit_testing }}
       - name: Build libOpenCOR
         if: ${{ (matrix.install_uninstall_and_package == 'ON') || (matrix.javascript_support == 'ON') || (matrix.code_analysis == 'ON') || (matrix.code_coverage == 'ON') || (matrix.memory_checks == 'ON') }}
         run: cmake --build build
@@ -439,10 +452,13 @@ jobs:
         run: cmake --build build --target install
       - name: Test libOpenCOR
         if: ${{ matrix.install_uninstall_and_package == 'ON' }}
+        shell: bash
         run: |
           cd tests/install
-          cmake -G Ninja -S . -B build
+          if [ '${{ matrix.use_install_prefix }}' == 'ON' ]; then export D_CMAKE_PREFIX_PATH='-DCMAKE_PREFIX_PATH=${{ github.workspace }}/install'; fi
+          cmake -G Ninja -S . -B build -DCMAKE_BUILD_TYPE=${{ matrix.build_type }} $D_CMAKE_PREFIX_PATH
           cmake --build build
+          if [ '${{ matrix.copy_dll }}' == 'ON' ]; then cp "C:/Program Files (x86)/libOpenCOR/bin/libOpenCOR.dll" build; fi
           build/testInstall
       - name: Uninstall libOpenCOR
         if: ${{ matrix.install_uninstall_and_package == 'ON' }}

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -435,9 +435,7 @@ jobs:
         run: cmake --build build
       - name: Code coverage
         if: ${{ matrix.code_coverage == 'ON' }}
-        run: |
-          cmake --build build --target ${{ matrix.target }}
-          if [ `ninja ${{ matrix.target }} | grep TOTAL | sed 's/ /\n/g' | grep "100.00%" | wc -l | sed 's/ //g'` -eq 4 ]; then exit 0; else exit 1; fi
+        run: if [ `cmake --build build --target ${{ matrix.target }} | grep TOTAL | sed 's/ /\n/g' | grep "100.00%" | wc -l | sed 's/ //g'` -eq 4 ]; then exit 0; else exit 1; fi
       - name: Code coverage report
         if: ${{ matrix.code_coverage == 'ON' }}
         run: cmake --build build --target ${{ matrix.target }}_report

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -122,7 +122,7 @@ jobs:
             target: unit_testing
             install_uninstall_and_package: ON
           - name: 'macOS static library (Intel)'
-            os: macos-14
+            os: macos-13
             build_type: Release
             code_analysis: OFF
             code_coverage: OFF
@@ -133,10 +133,9 @@ jobs:
             shared_libs: OFF
             unit_testing: ON
             target: unit_testing
-            target_arch: " -DTARGET_ARCHITECTURE=Intel"
             install_uninstall_and_package: ON
           - name: 'macOS shared library (Intel)'
-            os: macos-14
+            os: macos-13
             build_type: Release
             code_analysis: OFF
             code_coverage: OFF
@@ -147,7 +146,6 @@ jobs:
             shared_libs: ON
             unit_testing: ON
             target: unit_testing
-            target_arch: " -DTARGET_ARCHITECTURE=Intel"
             install_uninstall_and_package: ON
           - name: 'macOS static library (ARM)'
             os: macos-14
@@ -394,7 +392,7 @@ jobs:
           wget https://github.com/opencor/gha/releases/download/gha/doxygen.tar.gz -O - | tar -xz
           sudo mv doxygen /usr/local/bin
       - name: Configure libOpenCOR
-        run: ${{ matrix.context }} cmake -G Ninja -S . -B build ${{ matrix.target_arch }} -DBUILD_TYPE=${{ matrix.build_type }} -DCODE_ANALYSIS=${{ matrix.code_analysis }} -DCODE_COVERAGE=${{ matrix.code_coverage }} -DDOCUMENTATION=${{ matrix.documentation }} -DINSTALL_PREFIX=${{ github.workspace }}/install -DJAVASCRIPT_BINDINGS=${{ matrix.javascript_support }} -DJAVASCRIPT_UNIT_TESTING=${{ matrix.javascript_support }} -DMEMORY_CHECKS=${{ matrix.memory_checks }} -DPYTHON_BINDINGS=${{ matrix.python_support }} -DPYTHON_UNIT_TESTING=${{ matrix.python_support }} -DSHARED_LIBS=${{ matrix.shared_libs }} -DUNIT_TESTING=${{ matrix.unit_testing }}
+        run: ${{ matrix.context }} cmake -G Ninja -S . -B build -DBUILD_TYPE=${{ matrix.build_type }} -DCODE_ANALYSIS=${{ matrix.code_analysis }} -DCODE_COVERAGE=${{ matrix.code_coverage }} -DDOCUMENTATION=${{ matrix.documentation }} -DINSTALL_PREFIX=${{ github.workspace }}/install -DJAVASCRIPT_BINDINGS=${{ matrix.javascript_support }} -DJAVASCRIPT_UNIT_TESTING=${{ matrix.javascript_support }} -DMEMORY_CHECKS=${{ matrix.memory_checks }} -DPYTHON_BINDINGS=${{ matrix.python_support }} -DPYTHON_UNIT_TESTING=${{ matrix.python_support }} -DSHARED_LIBS=${{ matrix.shared_libs }} -DUNIT_TESTING=${{ matrix.unit_testing }}
       - name: Build libOpenCOR
         if: ${{ (matrix.install_uninstall_and_package == 'ON') || (matrix.javascript_support == 'ON') || (matrix.code_analysis == 'ON') || (matrix.code_coverage == 'ON') || (matrix.memory_checks == 'ON') }}
         run: cmake --build build

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -407,7 +407,7 @@ jobs:
           if [ '${{ matrix.use_install_prefix }}' == 'ON' ]; then export D_INSTALL_PREFIX='-DINSTALL_PREFIX=${{ github.workspace }}/install'; fi
           ${{ matrix.context }} cmake -G Ninja -S . -B build -DBUILD_TYPE=${{ matrix.build_type }} -DCODE_ANALYSIS=${{ matrix.code_analysis }} -DCODE_COVERAGE=${{ matrix.code_coverage }} -DDOCUMENTATION=${{ matrix.documentation }} $D_INSTALL_PREFIX -DJAVASCRIPT_BINDINGS=${{ matrix.javascript_support }} -DJAVASCRIPT_UNIT_TESTING=${{ matrix.javascript_support }} -DMEMORY_CHECKS=${{ matrix.memory_checks }} -DPYTHON_BINDINGS=${{ matrix.python_support }} -DPYTHON_UNIT_TESTING=${{ matrix.python_support }} -DSHARED_LIBS=${{ matrix.shared_libs }} -DUNIT_TESTING=${{ matrix.unit_testing }}
       - name: Build libOpenCOR
-        if: ${{ (matrix.install_uninstall_and_package == 'ON') || (matrix.javascript_support == 'ON') || (matrix.code_analysis == 'ON') || (matrix.code_coverage == 'ON') || (matrix.memory_checks == 'ON') }}
+        if: ${{ (matrix.install_uninstall_and_package == 'ON') || (matrix.code_analysis == 'ON') || (matrix.code_coverage == 'ON') || (matrix.memory_checks == 'ON') }}
         run: cmake --build build
       - name: Unit testing
         if: ${{ ((matrix.unit_testing == 'ON') && (matrix.code_analysis == 'OFF')) || (matrix.code_coverage == 'ON') || (matrix.memory_checks == 'ON') }}

--- a/cmake/packaging/patch.cmake.in
+++ b/cmake/packaging/patch.cmake.in
@@ -1,6 +1,17 @@
 file(GLOB_RECURSE CMAKE_FILE "@CMAKE_PROJECT_NAME@Targets.cmake")
 file(READ ${CMAKE_FILE} FILE_CONTENTS)
 
-string(REGEX REPLACE "\n[^\n]*INTERFACE_LINK_LIBRARIES[^\n]*\n" "\n" FILE_CONTENTS "${FILE_CONTENTS}")
+# By default, the INTERFACE_LINK_LIBRARIES entry lists all of libOpenCOR's dependencies, but we don't need them since we
+# embed them in our static version of libOpenCOR. However, on Windows and macOS, the INTERFACE_LINK_LIBRARIES entry
+# should list some needed system libraries. So, depending on the platform, we either set the INTERFACE_LINK_LIBRARIES
+# entry to those needed system libraries or remove the INTERFACE_LINK_LIBRARIES entry altogether.
+
+if(WIN32)
+    string(REGEX REPLACE "INTERFACE_LINK_LIBRARIES[^\n]*\n" "INTERFACE_LINK_LIBRARIES \"$<LINK_ONLY:crypt32.lib>;$<LINK_ONLY:version.lib>;$<LINK_ONLY:wldap32.lib>;$<LINK_ONLY:ws2_32.lib>\"\n" FILE_CONTENTS "${FILE_CONTENTS}")
+elseif(APPLE)
+    string(REGEX REPLACE "INTERFACE_LINK_LIBRARIES[^\n]*\n" "INTERFACE_LINK_LIBRARIES \"-framework CoreFoundation;-framework SystemConfiguration;$<LINK_ONLY:ldap>\"\n" FILE_CONTENTS "${FILE_CONTENTS}")
+else()
+    string(REGEX REPLACE "\n[^\n]*INTERFACE_LINK_LIBRARIES[^\n]*\n" "\n" FILE_CONTENTS "${FILE_CONTENTS}")
+endif()
 
 file(WRITE ${CMAKE_FILE} "${FILE_CONTENTS}")

--- a/tests/install/CMakeLists.txt
+++ b/tests/install/CMakeLists.txt
@@ -1,0 +1,13 @@
+cmake_minimum_required(VERSION 3.15)
+
+project(testInstall)
+
+set(CMAKE_CXX_STANDARD 20)
+
+find_package(libOpenCOR REQUIRED)
+
+add_executable(${PROJECT_NAME}
+    src/main.cpp
+)
+
+target_link_libraries(${PROJECT_NAME} PUBLIC libOpenCOR)

--- a/tests/install/bindings/python/test_install.py
+++ b/tests/install/bindings/python/test_install.py
@@ -1,0 +1,91 @@
+import libopencor as oc
+
+if __name__ == "__main__":
+    # Output the version of libOpenCOR.
+
+    print("---[ libOpenCOR version information ]---")
+    print("")
+    print(f"Version:        {oc.version()}")
+    print(f"Version string: {oc.version_string()}")
+
+    # Load a COMBINE archive from GitHub and output some information about it.
+
+    file = oc.File(
+        "https://raw.githubusercontent.com/opencor/libopencor/master/tests/res/cellml_2.omex"
+    )
+    document = oc.SedDocument(file)
+    simulation = document.simulations[0]
+
+    simulation.output_end_time = 5.0
+    simulation.number_of_steps = 500
+
+    instance = document.instantiate()
+
+    instance.run()
+
+    instance_task = instance.tasks[0]
+
+    print("")
+    print("---[ File information ]---")
+    print("")
+    print(f"URL: {file.url}")
+    print("Number of:")
+    print(f" - States and rates:    {instance_task.state_count}")
+    print(f" - Constants:           {instance_task.constant_count}")
+    print(f" - Computed constants:  {instance_task.computed_constant_count}")
+    print(f" - Algebraic variables: {instance_task.algebraic_count}")
+
+    # Output the results of the simulation.
+
+    print("")
+    print("---[ Simulation results ]---")
+    print("")
+    print(f"{instance_task.voi_name} ({instance_task.voi_unit})", end="")
+
+    for i in range(instance_task.state_count):
+        print(
+            f", {instance_task.state_name(i)} ({instance_task.state_unit(i)})", end=""
+        )
+
+    for i in range(instance_task.rate_count):
+        print(f", {instance_task.rate_name(i)} ({instance_task.rate_unit(i)})", end="")
+
+    for i in range(instance_task.constant_count):
+        print(
+            f", {instance_task.constant_name(i)} ({instance_task.constant_unit(i)})",
+            end="",
+        )
+
+    for i in range(instance_task.computed_constant_count):
+        print(
+            f", {instance_task.computed_constant_name(i)} ({instance_task.computed_constant_unit(i)})",
+            end="",
+        )
+
+    for i in range(instance_task.algebraic_count):
+        print(
+            f", {instance_task.algebraic_name(i)} ({instance_task.algebraic_unit(i)})",
+            end="",
+        )
+
+    print("")
+
+    for j in range(len(instance_task.voi)):
+        print(instance_task.voi[j], end="")
+
+        for i in range(instance_task.state_count):
+            print(f", {instance_task.state(i)[j]}", end="")
+
+        for i in range(instance_task.rate_count):
+            print(f", {instance_task.rate(i)[j]}", end="")
+
+        for i in range(instance_task.constant_count):
+            print(f", {instance_task.constant(i)[j]}", end="")
+
+        for i in range(instance_task.computed_constant_count):
+            print(f", {instance_task.computed_constant(i)[j]}", end="")
+
+        for i in range(instance_task.algebraic_count):
+            print(f", {instance_task.algebraic(i)[j]}", end="")
+
+        print("")

--- a/tests/install/src/main.cpp
+++ b/tests/install/src/main.cpp
@@ -1,0 +1,94 @@
+#include <iostream>
+#include <libopencor>
+
+int main()
+{
+    // Output the version of libOpenCOR.
+
+    std::cout << "---[ libOpenCOR version information ]---" << std::endl;
+    std::cout << std::endl;
+    std::cout << "Version:        " << libOpenCOR::version() << std::endl;
+    std::cout << "Version string: " << libOpenCOR::versionString() << std::endl;
+
+    // Load a COMBINE archive from GitHub and output some information about it.
+
+    auto file = libOpenCOR::File::create("https://raw.githubusercontent.com/opencor/libopencor/master/tests/res/cellml_2.omex");
+    auto document = libOpenCOR::SedDocument::create(file);
+    auto simulation = std::dynamic_pointer_cast<libOpenCOR::SedUniformTimeCourse>(document->simulations()[0]);
+
+    simulation->setOutputEndTime(5.0);
+    simulation->setNumberOfSteps(500);
+
+    auto instance = document->instantiate();
+
+    instance->run();
+
+    auto instanceTask = instance->tasks()[0];
+
+    std::cout << std::endl;
+    std::cout << "---[ File information ]---" << std::endl;
+    std::cout << std::endl;
+    std::cout << "URL: " << file->url() << std::endl;
+    std::cout << "Number of:" << std::endl;
+    std::cout << " - States and rates:    " << instanceTask->stateCount() << std::endl;
+    std::cout << " - Constants:           " << instanceTask->constantCount() << std::endl;
+    std::cout << " - Computed constants:  " << instanceTask->computedConstantCount() << std::endl;
+    std::cout << " - Algebraic variables: " << instanceTask->algebraicCount() << std::endl;
+
+    // Output the results of the simulation.
+
+    std::cout << std::endl;
+    std::cout << "---[ Simulation results ]---" << std::endl;
+    std::cout << std::endl;
+    std::cout << instanceTask->voiName() << " (" << instanceTask->voiUnit() << ")";
+
+    for (size_t i = 0; i < instanceTask->stateCount(); ++i) {
+        std::cout << "," << instanceTask->stateName(i) << " (" << instanceTask->stateUnit(i) << ")";
+    }
+
+    for (size_t i = 0; i < instanceTask->rateCount(); ++i) {
+        std::cout << "," << instanceTask->rateName(i) << " (" << instanceTask->rateUnit(i) << ")";
+    }
+
+    for (size_t i = 0; i < instanceTask->constantCount(); ++i) {
+        std::cout << "," << instanceTask->constantName(i) << " (" << instanceTask->constantUnit(i) << ")";
+    }
+
+    for (size_t i = 0; i < instanceTask->computedConstantCount(); ++i) {
+        std::cout << "," << instanceTask->computedConstantName(i) << " (" << instanceTask->computedConstantUnit(i) << ")";
+    }
+
+    for (size_t i = 0; i < instanceTask->algebraicCount(); ++i) {
+        std::cout << "," << instanceTask->algebraicName(i) << " (" << instanceTask->algebraicUnit(i) << ")";
+    }
+
+    std::cout << std::endl;
+
+    for (size_t j = 0; j < instanceTask->voi().size(); ++j) {
+        std::cout << instanceTask->voi()[j];
+
+        for (size_t i = 0; i < instanceTask->stateCount(); ++i) {
+            std::cout << "," << instanceTask->state(i)[j];
+        }
+
+        for (size_t i = 0; i < instanceTask->rateCount(); ++i) {
+            std::cout << "," << instanceTask->rate(i)[j];
+        }
+
+        for (size_t i = 0; i < instanceTask->constantCount(); ++i) {
+            std::cout << "," << instanceTask->constant(i)[j];
+        }
+
+        for (size_t i = 0; i < instanceTask->computedConstantCount(); ++i) {
+            std::cout << "," << instanceTask->computedConstant(i)[j];
+        }
+
+        for (size_t i = 0; i < instanceTask->algebraicCount(); ++i) {
+            std::cout << "," << instanceTask->algebraic(i)[j];
+        }
+
+        std::cout << std::endl;
+    }
+
+    return 0;
+}

